### PR TITLE
Add support for setting the 'defaultTableRecordsPerPageSelectOption' via a config file

### PR DESCRIPTION
**Pull request by ralphjsmit**
Currently, the `$defaultTableRecordsPerPageSelectOption` is set to 10 records per page. In almost all circumstances I'd prefer to show more records per page, e.g. like 50 or so.

In Discord I proposed some time ago to allow setting this via a config-file. There's not a perfect solution without breaking changes, but there is a pretty good solution.

Personally I'd prefer to make the `$defaultTableRecordsPerPageSelectOption` `null` by default. However, that would change the typehint `int` to `?int` and thus result in a breaking change for all the people who manually set a `protected int $defaultTableRecordsPerPageSelectOption = 999`.

The best solution I could come up with was setting the default value to `0`, which is still an integer. Zero records per page would never make sense, so that's a pretty good way to distinguish between values set by the user and the default value.

### CODE REVIEW

**Comment by ralphjsmit** [comment]
PS: is it an idea to also move the following values to a configuration option?

```php
protected function getTableRecordsPerPageSelectOptions(): array
{
    return [5, 10, 25, 50];
}
```

File: packages/admin/config/filament.php [hunk 1]
```diff
@@ -211,6 +211,7 @@
             'actions' => [
                 'type' => Action::class,
             ],
+            'default_table_records_per_page_select_option' => 10,
```
Comment (danharrin) [inline packages/admin/config/filament.php:214]: This belongs in the tables config, `tables.pagination.default_records_per_page`

File: packages/tables/src/Concerns/CanPaginateRecords.php [hunk 2]
```diff
@@ -15,7 +15,7 @@ trait CanPaginateRecords
 
     public $tableRecordsPerPage;
 
-    protected int $defaultTableRecordsPerPageSelectOption = 10;
+    protected int $defaultTableRecordsPerPageSelectOption = 0;
```
Comment (danharrin) [inline packages/tables/src/Concerns/CanPaginateRecords.php:18]:
```suggestion
    protected ?int $defaultTableRecordsPerPageSelectOption = null;
```

File: packages/tables/src/Concerns/CanPaginateRecords.php [hunk 3]
```diff
@@ -50,7 +50,10 @@ protected function getTableRecordsPerPageSelectOptions(): array
 
     protected function getDefaultTableRecordsPerPageSelectOption(): int
     {
-        $perPage = session()->get($this->getTablePerPageSessionKey(), $this->defaultTableRecordsPerPageSelectOption);
+        $perPage = session()->get(
+            $this->getTablePerPageSessionKey(),
+            $this->defaultTableRecordsPerPageSelectOption ?: config('filament.layout.tables.default_table_records_per_page_select_option')
```
Comment (danharrin) [inline packages/tables/src/Concerns/CanPaginateRecords.php:55]:
```suggestion
            $this->defaultTableRecordsPerPageSelectOption ?? config('filament.layout.tables.default_table_records_per_page_select_option')
```

File: packages/admin/config/filament.php (hunk 1 above)
Comment (ralphjsmit) [reply packages/admin/config/filament.php]: Done!

File: packages/tables/src/Concerns/CanPaginateRecords.php (hunk 2 above)
Comment (ralphjsmit) [reply packages/tables/src/Concerns/CanPaginateRecords.php:18]: Are you sure? This is a breaking change for people who did:

```php
protected int $defaultTableRecordsPerPageSelectOption = 50;
```

Comment (danharrin) [reply packages/tables/src/Concerns/CanPaginateRecords.php:18]: Doesn't PHP allow this, since `int` is a subset of `?int` ? testing this would be great

Comment (ralphjsmit) [reply packages/tables/src/Concerns/CanPaginateRecords.php:18]: Schermafbeelding 2022-05-26 om 22 12 31

Comment (ralphjsmit) [reply packages/tables/src/Concerns/CanPaginateRecords.php:18]: Good idea!

**Pull request merged**